buffer.go

```diff
@@ -189,41 +179,40 @@ func (b *Buffer) tell() int {
 		seeker := b.reader.(io.Seeker)
 		off, _ := seeker.Seek(0, io.SeekCurrent)
 
-		return int(off) + (b.bitIndex >> 3) - b.length
+		return int(off) + (b.bitIndex >> 3) - len(b.bytes)
 	}
 
 	return b.bitIndex >> 3
 }
 
 func (b *Buffer) discardReadBytes() {
 	bytePos := b.bitIndex >> 3
-	if bytePos == b.length {
-		b.bytes.Truncate(0)
+	if bytePos == len(b.bytes) {
```

Comment:
Is this logic correct by the way? What if bitIndex is 9 and length (len(b.bytes)) is 1? Isn't 1bit discarded unexpectedly?

Reply:
The logic is rewritten from https://github.com/phoboslab/pl_mpeg/blob/master/pl_mpeg.h#L1553.

Reply:
So, is the original logic correct...? This seems a little suspicious

Reply:
Aha, now I think this is correct: when `bytePos == len(b.bytes)`, the lower 3bits of `bitIndex` must be 0, or `bitIndex` would exceed the buffer length.